common/c/targets/esp8266/driver/gpio.h

```diff
@@ -95,13 +95,13 @@ typedef struct {
 } GPIO_ConfigTypeDef;
 
 #define GPIO_OUTPUT_SET(gpio_no, bit_value) do { \
-    gpio_output_conf((bit_value)<<(gpio_no), ((~(bit_value))&0x01)<<(gpio_no), 1<<(gpio_no), 0) \
-    while(0);
+    gpio_output_conf((bit_value)<<(gpio_no), ((~(bit_value))&0x01)<<(gpio_no), 1<<(gpio_no), 0); \
+    } while(0);
```

Comment:
do-while(0) should be used as follows. (Note the absence of trailing semicolon.)
```c
do {
    ...
} while (0)
```

Furthermore, do-while is actually not needed for single function invocation. The next would be enough:
```c
#define GPIO_OUTPUT_SET(gpio_no, bit_value) \
    gpio_output_conf((bit_value)<<(gpio_no), ((~(bit_value))&0x01)<<(gpio_no), 1<<(gpio_no), 0)
```

Also, add spaces around operators.